File: CSharpFundamentals/CSharpOOPbasics/InterfacesAndAbstractionEx/Ferrari/Ferrari.cs
using System.Text;

namespace Ferrari
{
    public class Ferrari : ICar
    {
        private const string MODEL = "488-Spider";
        private const string STOP = "Brakes!";
        private const string START = "Zadu6avam sA!";

        public string Model { get; private set; }
        public string Driver { get; private set; }

        public Ferrari(string driver)
        {
            this.Driver = driver;
            this.Model = MODEL;
        }

        public string brake()
        {
            return STOP;
        }

        public string gas()
        {
            return START;
        }

        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();
            sb.Append($"{Model}/{brake()}/{gas()}/{Driver}");
            return sb.ToString().TrimEnd();
        }
    }
}

File: JavaFundamentals/JavaOOP/Interfaces-And-Abstraction/src/ferrari/Ferrari.java
package ferrari;

public class Ferrari implements Car {
    private static final String MODEL = "488-Spider";
    private String driverName;

    public Ferrari(String driverName) {
        this.driverName = driverName;
    }

    @Override
    public String brakes() {
        return "Brakes!";
    }

    @Override
    public String gas() {
        return "Zadu6avam sA!";
    }

    @Override
    public String toString() {
        return String.format("%s/%s/%s/%s",
                MODEL,
                this.brakes(),
                this.gas(),
                this.driverName);
    }
}
